lib/src/native/realm_core.dart

```diff
@@ -326,6 +326,47 @@ class _RealmCore {
     return RealmResultsHandle._(pointer);
   }
 
+  RealmResultsHandle queryTable(Realm realm, int classKey, String query, List<Object> args) {
+    return using((arena) {
+      final length = args.length;
+      final argsPointer = arena<Pointer<realm_value_t>>(length);
```

Comment:
```suggestion
      final argsPointer = arena<realm_value_t>(length);
```

Reply:
arena<Pointer<realm_value_t>>(length) will allocate `length` number of pointers to realm_value_t, but we want `length` number of realm_value_t in memory.

Reply:
I get it. A bit puzzled why my test worked.